### PR TITLE
Add pwm support for mega2560

### PR DESCRIPTION
**Pull request by sepotvin**
Add support for the pwm (6 timers) of atmega2560 chips and mega2560 board. It also fixes a typo in the atmega328p which was disconnecting the wrong port.

### CODE REVIEW

File: chips/atmega2560-hal/src/pwm.rs [hunk 1]
```diff
@@ -0,0 +1,363 @@
...
+                },
+            },
+        },
+    }
+}
+
+avr_hal::impl_pwm! {
+    /// Use `TC1` for PWM (pins `PB5`, `PB6`)
+    ///
+    /// Note: `PB7` is not currently supported (enter in conflict with `PB7` from timer0).
```
Comment (Rahix) [inline chips/atmega2560-hal/src/pwm.rs:91]: You can support one pin from multiple timers; take a look at `atmega32u4-hal` where this is done, for example.  You'll have one method `.into_pwm()` for the default timer that you think most people would want to use and a secondary `.into_pwm1()` for the alternative (or `.into_pwm(whatever timer number)()`).

Please also mention this in the overview table as is done in `atmega32u4-hal`.

**Review by Rahix — COMMENTED** [review]
Hey, thanks a lot for working on this!  I have one minor thing I could see in the diff, I left a comment.  I don't have this board on hand to test whether everything works, so I'll have to ask: Did you give all pins a quick try whether the waveform looks okay?

File: chips/atmega2560-hal/src/pwm.rs (hunk 1 above)
Comment (sepotvin) [reply chips/atmega2560-hal/src/pwm.rs:91]: Thanks for the pointer, I'll have a look and submit an update

File: boards/arduino-mega2560/src/lib.rs [hunk 2]
```diff
@@ -17,3 +17,53 @@ pub use crate::pins::*;
...
+/// | `PE4` | `.into_pwm(&mut timer3)` |
+/// | `PE5` | `.into_pwm(&mut timer3)` |
+/// | `PG5` | `.into_pwm(&mut timer0)` |
+/// | `PH3` | `.into_pwm(&mut timer4)` |
+/// | `PH4` | `.into_pwm(&mut timer4)` |
+/// | `PH5` | `.into_pwm(&mut timer4)` |
+/// | `PH6` | `.into_pwm(&mut timer2)` |
+/// | `PL3` | `.into_pwm(&mut timer5)` |
+/// | `PL4` | `.into_pwm(&mut timer5)` |
+/// | `PL5` | `.into_pwm(&mut timer5)` |
```
Comment (couchand) [inline boards/arduino-mega2560/src/lib.rs:64]: Perhaps here the board pin numbers should be added or used instead?

Comment (Rahix) [reply boards/arduino-mega2560/src/lib.rs:64]: For Arduino Uno & Leonardo, I have created two versions of this table:  In the HAL crate (`atmega328p-hal`/`atmega32u4-hal`) the table uses MCU pin names and in the BSP crate (`arduino-uno`/`arduino-leonardo`) is uses board pin names.  I would suggest doing the same here.

**Review by Rahix — APPROVED** [review]
Looks good! I've taken the `ATmega328P` patch and applied that separately already.

**Pull request merged**